Samples/excel-add-in-shape-dashboard/src/taskpane/taskpane.ts

```diff
@@ -0,0 +1,436 @@
+
+    await context.sync();
+    showStatus("Add sample data - success!", false);
+  });
+}
+
+async function createDashboard() {
+  await Excel.run(async (context) => {
+    showStatus("Running...", false);
+    let shapes = context.workbook.worksheets.getItem("Sample").shapes;
+    let array = [];
+
+    let region = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    region.left = 550;
+    region.top = 100;
+    region.width = 100;
+    region.height = 100;
+    array.push(region);
+
+    let state = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    state.left = 700;
+    state.top = 100;
+    state.width = 100;
+    state.height = 100;
+    array.push(state);
+
+    let line = shapes.addGeometricShape(Excel.GeometricShapeType.rectangle);
+    line.left = 500;
+    line.top = 230;
+    line.width = 350;
+    line.height = 1;
+    array.push(line);
+
+    let category = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    category.left = 550;
+    category.top = 260;
+    category.width = 100;
+    category.height = 100;
+    array.push(category);
+
+    let subCategory = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    subCategory.left = 700;
+    subCategory.top = 260;
+    subCategory.width = 100;
+    subCategory.height = 100;
+    array.push(subCategory);
+
+    let line2 = shapes.addGeometricShape(Excel.GeometricShapeType.rectangle);
+    line2.left = 500;
+    line2.top = 390;
+    line2.width = 350;
+    line2.height = 1;
+    array.push(line2);
+
+    let maxScale = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    maxScale.left = 550;
+    maxScale.top = 420;
+    maxScale.width = 100;
+    maxScale.height = 100;
+    array.push(maxScale);
+
+    let sumScale = shapes.addGeometricShape(Excel.GeometricShapeType.roundRectangle);
+    sumScale.left = 700;
+    sumScale.top = 420;
+    sumScale.width = 100;
+    sumScale.height = 100;
+    array.push(sumScale);
+
+    shapes.addGroup(array);
+    await context.sync();
+
+    showStatus("Create empty dashboard - success!", false);
+  });
+}
+
+async function addInformation() {
+  await Excel.run(async (context) => {
+    showStatus("Running...", false);
+    let shapes = context.workbook.worksheets.getItem("Sample").shapes;
+    shapes.load("items");
+    await context.sync();
+
+    let shapeGroup = shapes.items[0].group.shapes;
+    shapeGroup.load("items");
+    await context.sync();
+
+    let text2 = context.workbook.worksheets.getItem("Sample").getRange("E152:E152");
+    text2.load("values");
+    await context.sync();
+
+    let region = shapeGroup.items[0];
+    region.textFrame.textRange.text = Math.ceil(parseFloat(text2.values.toString())).toString() + "\nRegions";
+    region.fill.foregroundColor = "#339933";
+
+    let text3 = context.workbook.worksheets.getItem("Sample").getRange("E157:E157");
+    text3.load("values");
```

Comment:
This is definitely an Excel bug, but if you only have one state in this column, the text displays 0.999999

![image](https://github.com/user-attachments/assets/0e3e6202-e85c-4346-bcf8-50c5e3b0393e)